addons/html_editor/static/src/core/input_plugin.js

```diff
@@ -11,14 +12,34 @@ export class InputPlugin extends Plugin {
         this.addDomListener(this.editable, "input", this.onInput);
     }
 
+    ignoreInput() {
+        const selection = this.document.getSelection();
+        const ignoreElementForInput = (el) => {
+            if (!el) {
+                return false;
+            }
+            return isProtected(el) || isProtecting(el);
+        };
+        return (
+            ignoreElementForInput(selection.anchorNode) ||
+            ignoreElementForInput(selection.focusNode)
+        );
+    }
+
     onBeforeInput(ev) {
         this.dispatch("HISTORY_STAGE_SELECTION");
+        if (this.ignoreInput()) {
```

Comment:
Can you use the target instead of the selection? As the code should be easier, maybe we don't need a function. If we keep a function, call it `isTargetProtected`?